File: Path Sum - Leetcode 112/Path Sum - Leetcode 112.js
function hasPathSum(root, targetSum) {
    return hasSum(root, 0, targetSum);
}

function hasSum(node, currentSum, targetSum) {
    if (node === null) return false;

    currentSum += node.val;

    if (node.left === null && node.right === null) {
        return currentSum === targetSum;
    }

    return hasSum(node.left, currentSum, targetSum) || hasSum(node.right, currentSum, targetSum);
}

File: Path Sum - Leetcode 112/Path Sum - Leetcode 112.java
public class Solution {
    public boolean hasPathSum(TreeNode root, int targetSum) {
        return hasSum(root, 0, targetSum);
    }

    private boolean hasSum(TreeNode node, int currentSum, int targetSum) {
        if (node == null) return false;

        currentSum += node.val;

        if (node.left == null && node.right == null) {
            return currentSum == targetSum;
        }

        return hasSum(node.left, currentSum, targetSum) || hasSum(node.right, currentSum, targetSum);
    }
}
